users_ldap_push/models/res_users.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    This module copyright (C) 2015 Therp BV (<http://therp.nl>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import ldap
+import ldap.modlist
+from openerp import _, models, fields, api, exceptions
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    ldap_entry_dn = fields.Char('LDAP DN', readonly=True)
+    is_ldap_user = fields.Boolean(
+        'LDAP user', compute='_compute_is_ldap_user', default=True)
+
+    @api.model
+    @api.returns('self', lambda record: record.id)
+    def create(self, values):
+        result = super(ResUsers, self).create(values)
+        result.push_to_ldap(values)
+        return result
+
+    @api.multi
+    def write(self, values):
+        result = super(ResUsers, self).write(values)
+        self.push_to_ldap(values)
+        return result
+
+    @api.multi
+    def _push_to_ldap_possible(self, values):
+        return bool(self._get_ldap_configuration())
+
+    @api.multi
+    def _get_ldap_configuration(self):
+        self.ensure_one()
+        return self.sudo().company_id.ldaps.filtered('create_ldap_entry')[:1]
+
+    @api.multi
+    def _get_ldap_values(self, values):
+        self.ensure_one()
```

Comment:
are those calls expensive? I try to develop the habit to say `ensure_one` on every `@api.multi` function that makes use of assignments in order to see at one glace what kind of method it is